src/index.ts

```diff
@@ -41,13 +41,14 @@ const getOptionalInput = (name: string) => core.getInput(name) || undefined;
         "No changesets found, attempting to publish any unpublished packages to npm"
       );
 
-      let npmrcPath = `${process.env.HOME}/.npmrc`;
-      if (fs.existsSync(npmrcPath)) {
+      let npmrcHomePath = `${process.env.HOME}/.npmrc`;
+      let npmrcProjectRootPath = `${process.env.PWD}/.npmrc`;
```

Comment:
is there any particular reason why you have chosen this over `process.cwd()`?

Reply:
`process.env.PWD` is a safer way compared to `process.env.cwd()` because while your process is running and if one of the commands in that process changes the working directory while it's still executing the value of `process.env.cwd()` will change at all the places where it's being referred. `process.env.PWD` guarantees that your current working directory will be set to a path when the command actually started executing and wouldn't change during the lifecycle of the process